### PR TITLE
feat: add protocol server

### PR DESCRIPTION
**Pull request by LingKa28**
Please briefly answer these questions:

* what problem are you trying to solve? (or if there's no problem, what's the motivation for this change?)
Add new feature.

* what changes does this pull request make?
  * add protocol server

* are there any non-obvious implications of these changes? (does it break compatibility with previous versions, etc)
No.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	"google.golang.org/grpc"
+
+	curppb "github.com/xline-kv/mock-xline/gen/curppb"
+	"github.com/xline-kv/mock-xline/sample"
+	"github.com/xline-kv/mock-xline/server"
+)
+
+func main() {
+	name := flag.String("name", "node1", "node name")
+	addr := flag.String("addr", "127.0.0.1:5555", "node address")
```
Comment (bsbds) [inline main.go]: Using the default port 2379 could be better.

File: server/protocol.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+	return &protocolServer{
+		fetchClusterSample: fetchClusterSample,
+		proposeSample:      proposeSample,
+		waitSyncSample:     waitSyncSample,
+		commandSample:      sample.CommandMap{},
+	}
+}
+
+func (s *protocolServer) FetchCluster(ctx context.Context, req *curppb.FetchClusterRequest) (*curppb.FetchClusterResponse, error) {
+	fmt.Println("fetch cluster", req)
```
Comment (bsbds) [inline server/protocol.go:38]: Could using logging be better than println?

File: server/protocol.go [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+	res, ok := s.proposeSample[cmd.String()]
+	if !ok {
+		return nil, status.Errorf(codes.NotFound, "data not found")
+	}
+
+	return res, nil
+}
+
+func (s *protocolServer) WaitSynced(ctx context.Context, req *curppb.WaitSyncedRequest) (*curppb.WaitSyncedResponse, error) {
+	time.Sleep(50 * time.Millisecond)
```
Comment (bsbds) [inline server/protocol.go]: Please define a constant for this.

Comment (bsbds) [reply server/protocol.go]: Also, should we add latency for each seperate request? Currently all requests have the same lantency.

File: sample/protocol.go [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+	var leaderId serverId = 1
+	res := &curppb.FetchClusterResponse{
+		LeaderId:  &leaderId,
+		Term:      1,
+		ClusterId: 1,
+		Members: []*curppb.Member{
+			{
+				Id:    1,
+				Name:  "node1",
+				Addrs: []string{"127.0.0.1:5555"},
```
Comment (bsbds) [inline sample/protocol.go]: same as above

File: main.go (hunk 1 above)
Comment (LingKa28) [reply main.go]: Done.

File: server/protocol.go (hunk 3 above)
Comment (LingKa28) [reply server/protocol.go]: We can use it as a configuration item in the later pr.

File: sample/protocol.go (hunk 4 above)
Comment (LingKa28) [reply sample/protocol.go]: Done.

File: server/protocol.go (hunk 2 above)
Comment (LingKa28) [reply server/protocol.go:38]: We plan to assert the output in a later update. So use fmt for output temporary

**Review by Phoenix500526 — COMMENTED** [review]
We need some examples to show the usage of the mock server. Additionally, we need to invert control, allowing the client to determine the response for a given request, rather than hard-coding it in the mock server. Please create issues to track these tasks.

**Comment by LingKa28** [comment]
I've added issue to track the problem.

**Review by Phoenix500526 — APPROVED** [review]
LGTM

**Review by bsbds — APPROVED** [review]

**Pull request merged**